File: CompilerRR/CompilerRR/src/ch/fhnw/cpib/compiler/classes/StorageDeclaration.java
package ch.fhnw.cpib.compiler.classes;

import ch.fhnw.cpib.compiler.cst.interfaces.IConcSyn;
import ch.fhnw.cpib.compiler.cst.interfaces.IConcSyn.IOptionalCHANGEMODE;
import ch.fhnw.cpib.compiler.cst.interfaces.IConcSyn.ITypedIdent;

public class StorageDeclaration implements IConcSyn.IStorageDeclaration {

	IConcSyn.IOptionalCHANGEMODE mode;
	IConcSyn.ITypedIdent ident;
	public StorageDeclaration(IOptionalCHANGEMODE mode, ITypedIdent ident) {
		super();
		this.mode = mode;
		this.ident = ident;
	}
	
	
}

File: CompilerMR/IML_Compiler/IMLParser/DataTypes/Concrete/StorageDeclaration.cs
using IML_Compiler.IMLParser.DataTypes.Abstract;

namespace IML_Compiler.IMLParser.DataTypes.Concrete
{
    class StorageDeclaration : IStorageDeclaration
    {
        public IOptionalChangeMode OptionalChangeMode { get; private set; }
        public ITypedIdent TypedIdent { get; private set; }

        public StorageDeclaration (IOptionalChangeMode mode, ITypedIdent ident)
        {
            OptionalChangeMode = mode;
            TypedIdent = ident;
        }

        public AbsStorageDeclaration ToAbySyn()
        {
            var abs = TypedIdent.ToAbySyn();
            return new AbsStorageDeclaration()
            {
                ChangeMode = OptionalChangeMode.ToAbySyn(),
                Ident = abs.Ident,
                TypeDeclaration = abs.Type
            };
        }
    }
}
